setup.cfg

```diff
@@ -45,3 +40,141 @@ ignore = E203,E501
 # D213 Multi-line docstring summary should start at the second line (reason: pep257 default)
 ignore = D107, D203, D213
 ignore_decorators = overrides
+
+[tox:tox]
+min_version = 4.0
+env_list =
+    # Parametrized environments.
+    # First parameter allows us to choose Python 3.8 or 3.10.
+    # Second parameter chooses how to define the environment:
+    #  withreq: using requirements-devel.txt
+    #  noreq: Without either requirements file (but including dev requirements)
```

Comment:
what's the reasoning behind this 'noreq' env? Did this concept exist before?

Reply:
It was helpful for me while implementing tox to figure out why some things were breaking, so it's new coming from that. By the time I'd figured out the issues, having it had become sufficiently useful that I kept it around.